source/material_model/reactive_fluid_transport.cc

```diff
@@ -211,84 +218,92 @@ namespace aspect
     {
       base_model->evaluate(in,out);
 
-      const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
-
-      // Modify the viscosity from the base model based on the presence of fluid.
-      if (in.requests_property(MaterialProperties::viscosity))
+      if (fluid_solid_reaction_scheme != katz2003)
         {
-          // Scale the base model viscosity value based on the porosity.
-          for (unsigned int q=0; q<out.n_evaluation_points(); ++q)
+          const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+          // Modify the viscosity from the base model based on the presence of fluid.
+          if (in.requests_property(MaterialProperties::viscosity))
             {
-              const double porosity = std::max(in.composition[q][porosity_idx],0.0);
-              out.viscosities[q] *= (1.0 - porosity) * exp(- alpha_phi * porosity);
+              // Scale the base model viscosity value based on the porosity.
+              for (unsigned int q=0; q<out.n_evaluation_points(); ++q)
+                {
+                  const double porosity = std::max(in.composition[q][porosity_idx],0.0);
+                  out.viscosities[q] *= (1.0 - porosity) * exp(- alpha_phi * porosity);
```

Comment:
@naliboff This is actually slightly different from what the Katz2003 model does. If everything was the same, it would be super easy to combine. But I agree, it would be nice not having to duplicate this.